Doc/library/os.rst

```diff
@@ -2292,6 +2292,8 @@ features:
    will fail with an :exc:`OSError` subclass in a number of cases:
 
    On Windows, if *dst* exists a :exc:`FileExistsError` is always raised.
+   The operation may fail if *src* and *dst* are on different filesystems. Use
+   :func:`shutil.move` to support moves to a different filesystem.
```

Comment:
I believe since this statement is Windows-specific, it's fine to keep it under the Windows section despite being slightly unrelated to the general case of *dst* existing. But let me know if it's preferred to open up a new paragraph above/below (or if removing the sentence on *dst* exists is preferred).